pkg/helm/findscore.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2021 FairwindsOps Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"strings"
+
+	"github.com/fairwindsops/nova/pkg/output"
+	version "github.com/mcuadros/go-version"
+	"helm.sh/helm/v3/pkg/release"
+
+	"k8s.io/klog"
+)
+
+// FindBestArtifactHubMatch takes the helm releases found in the cluster and attempts to match those to a package in artifacthub
+func FindBestArtifactHubMatch(clusterRelease *release.Release, ahubPackages []ArtifactHubHelmPackage) *output.ReleaseOutput {
+	var highScore int
+	var highScorePackage ArtifactHubHelmPackage
+	for _, p := range ahubPackages {
+		score := 0
+		if p.Name != clusterRelease.Chart.Metadata.Name {
+			continue
+		}
+		score = scoreChartSimilarity(clusterRelease, p)
+		if score > highScore {
+			highScore = score
+			highScorePackage = p
+		}
+	}
+	klog.V(10).Infof("highScore for '%s': %d, highScorePackage Repo: %s", clusterRelease.Chart.Metadata.Name, highScore, highScorePackage.Repository.Name)
+	return prepareOutput(clusterRelease, highScorePackage)
+}
+
+func prepareOutput(release *release.Release, pkg ArtifactHubHelmPackage) *output.ReleaseOutput {
+	return &output.ReleaseOutput{
+		ReleaseName: release.Name,
+		ChartName:   release.Chart.Metadata.Name,
+		Namespace:   release.Namespace,
+		Description: release.Chart.Metadata.Description,
+		Home:        release.Chart.Metadata.Home,
+		Icon:        release.Chart.Metadata.Icon,
+		Installed: output.VersionInfo{
+			Version:    release.Chart.Metadata.Version,
+			AppVersion: release.Chart.Metadata.AppVersion,
+		},
+		Latest: output.VersionInfo{
+			Version:    pkg.Version,
+			AppVersion: pkg.AppVersion,
+		},
+		IsOld:       version.Compare(release.Chart.Metadata.Version, pkg.Version, "<"),
+		Deprecated:  pkg.Deprecated,
+		HelmVersion: "3",
+	}
+}
+
+func scoreChartSimilarity(release *release.Release, pkg ArtifactHubHelmPackage) int {
+	ret := 0
+	var preferredRepositories = []string{
+		"bitnami",
+		"fairwinds-stable",
+		"ingress-nginx",
```

Comment:
Should we add jetstack to this one too? I feel like they definitely make the list :-D

Reply:
Looks like the repo for cert-manager is called cert-manager in artifact hub so I'll add that.

Funny enough the `bitnami` repo also has a cert-manager chart, but the `cert-manager` repo should still win out in this code because it's both `official` and `verified` + adding preferred here. https://artifacthub.io/packages/helm/cert-manager/cert-manager